### PR TITLE
Replace askLogFunc with askLoggerIO

### PR DESCRIPTION
**Pull request by mitchellvitez**
Just trying to do a small FIXME here

no haddock changes

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: persistent-postgresql/Database/Persist/Postgresql.hs [hunk 1]
```diff
@@ -186,7 +186,7 @@ createPostgresqlPool = createPostgresqlPoolModified (const $ return ())
 --
 -- @since 2.1.3
 createPostgresqlPoolModified
-    :: (MonadUnliftIO m, MonadLogger m)
+    :: (MonadUnliftIO m, MonadLoggerIO m)
```
Comment (parsonsmatt) [inline persistent-postgresql/Database/Persist/Postgresql.hs:189]: Changing the signature of these functions is a breaking change and would require a major version bump.

File: persistent/ChangeLog.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## 2.11.0.1
 
+* [#1162](https://github.com/yesodweb/persistent/pull/1162)
+  * Replace `askLogFunc` with `askLoggerIO`
```
Comment (parsonsmatt) [inline persistent/ChangeLog.md:6]: 2.11.0.1 has already been released on Hackage - since this is a breaking change, it'd need to go under 2.12 or 3.0.

**Review by parsonsmatt — COMMENTED** [review]
Thanks for the PR! I think it'd be best if we made an issue for this and tagged it in for the next major version bump. It doesn't seem like a large enough change to release all on it's own right now.

**Comment by mitchellvitez** [comment]
>best if we made an issue for this and tagged it in for the next major version bump. It doesn't seem like a large enough change to release all on it's own right now.

Agreed. To do that, do I just need to put this change under 2.12 in the changelog?

Also wasn't sure how to get CI building, looks like a github actions thing

**Comment by parsonsmatt** [comment]
I fixed CI on  master, try merging that in.

I'm hesitant to release 2.12 with only this change, and I'd prefer to batch it up in the next breaking release.

**Pull request merged**